my-dsl.md

```diff
@@ -1,38 +1,105 @@
 # Language
-_What is the name of the language? Link the name to its webpage 
+_What is the name of the language? Link the name to its webpage
 (if appropriate)._
 
+[Dockerfile Command Language](https://docs.docker.com/engine/reference/builder/)
 
 # Domain
 _Describe the language's domain in five words._
 
+Efficient portable software runtime environments.
 
 # Computational model
-_We don't yet have a great definition of the term "computational model". 
-For now, try to come up with the clearest, most concise explanation of 
+_We don't yet have a great definition of the term "computational model".
+For now, try to come up with the clearest, most concise explanation of
 what happens when a program in your DSL runs._
 
+A user describes the overall build of a container&mdash;a sort of lightweight
+virtual appliance&mdash;in a text file called a Dockerfile with limited text
+commands. When a user builds an image from the Dockerfile, the text commands are
+parsed and with each step a partial container is built representing the state of
+the machine&mdash;the Docker commands are converted to lower-level system
+commands as needed. After the final command, the user can use the built machine
+just as they would a virtual appliance without worrying about the intermediate
+steps. If the Dockerfile is edited, the image builder will find the modified
+line, and only rebuild and commit images that came from commands with and after
+that line.
 
 # DSL-ness
-_Fowler writes about a spectrum of languages, from general-purpose languages to 
-"purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
-and why?_ 
+_Fowler writes about a spectrum of languages, from general-purpose languages to
+"purely" domain-specific. Where does the DSL you chose fall on this spectrum,
+and why?_
 
+The Dockerfile Command Langauge is "purely" domain-specific. It serves one
+purpose: to describe and build reproducible Docker containers/images. While it
+looks a bit like a scripting language, the commands are extremely limited for
+the purposes of readability, reproducibility, and commitability. The langauge
+itself is certainly not Turing-complete&mdash;Fowler's litmus test for generic
+DSLs.
+
+While I definitely think the Dockerfile's commands are a DSL&mdash;the
+documentation even goes as far as refering to it as a DSL&mdash;it verges on the
+line of simply being a configuration file which Fowler would likely not count
+as a DSL. The reason I'd argue it is not simply a configuration file is there
+needs to be a logical ordering of steps that are user determined.
 
```

Comment:
I didn't mean to imply that its ordering alone makes it a DSL, but I think it is a differentiating factor between a Dockerfile set of commands and a simple configuration file. In my eyes, it's just one test (of many) to distinguish the two.